state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -279,7 +267,10 @@ impl L1Fetcher {
                             }
                         } else {
                             tracing::debug!("Cannot get latest block number...");
-                            tokio::time::sleep(Duration::from_secs(LONG_POLLING_INTERVAL_S)).await;
+                            tokio::select! {
```

Comment:
I really think this code should be encapsulated - maybe into some wrapper of `CancellationToken`, as suggested in #eqg-lang-rust ...

Reply:
Done.